assessment/collectors/project_analyzer/file_dependency_analyzer.go

```diff
@@ -0,0 +1,297 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.*/
+
+package assessment
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"golang.org/x/tools/go/packages"
+)
+
+// DependencyAnalyzer defines the interface for dependency analysis
+type DependencyAnalyzer interface {
+	GetDependencyGraph(directory string) map[string]map[string]struct{}
+	IsDAOClass(filePath string, fileContent string) bool
+	GetExecutionOrder(projectDir string) (map[string]map[string]struct{}, [][]string)
+}
+
+// BaseAnalyzer provides default implementation for execution order
+type BaseAnalyzer struct{}
+
+// GoDependencyAnalyzer implements DependencyAnalyzer for Go projects
+type GoDependencyAnalyzer struct {
+	BaseAnalyzer
+}
+
+func (g *GoDependencyAnalyzer) GetDependencyGraph(directory string) map[string]map[string]struct{} {
+
+	fmt.Println(directory)
+	cfg := &packages.Config{
+		Mode: packages.NeedName | packages.NeedFiles | packages.NeedSyntax | packages.NeedTypes | packages.NeedTypesInfo,
+		Dir:  (directory),
+	}
+
+	fmt.Println(directory)
+
+	pkgs, err := packages.Load(cfg, "./...")
+
+	fmt.Println(directory)
+	if err != nil {
+		log.Fatalf("Error loading packages: %v", err)
+	}
+
+	// Dependency graph: key = file, value = list of files it depends on
+	dependencyGraph := make(map[string]map[string]struct{})
+
+	// Iterate through all packages and process their files
+	for _, pkg := range pkgs {
+		if pkg.TypesInfo == nil {
+			continue
+		}
+
+		// Process symbol usages (functions, variables, structs)
+		for ident, obj := range pkg.TypesInfo.Uses {
+			if obj != nil && obj.Pos().IsValid() {
+				useFile := pkg.Fset.Position(ident.Pos()).Filename
+
+				// Only process files inside the project directory
+				if strings.HasPrefix(useFile, directory) {
+					// Get the file where the symbol is defined
+					defFile := pkg.Fset.Position(obj.Pos()).Filename
+
+					// Only add if the file is inside the project directory and avoid redundant edges
+					if strings.HasPrefix(defFile, directory) && useFile != defFile {
+						// Initialize the map for the useFile if not present
+						if _, ok := dependencyGraph[useFile]; !ok {
+							dependencyGraph[useFile] = make(map[string]struct{})
+						}
+
+						if _, ok := dependencyGraph[defFile]; !ok {
+							dependencyGraph[defFile] = make(map[string]struct{})
+						}
+
+						// Add the dependency edge only if it doesn't already exist
+						if _, exists := dependencyGraph[useFile][defFile]; !exists {
+							dependencyGraph[useFile][defFile] = struct{}{}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	// Print the dependency graph
+	fmt.Println("\nDependency Graph:")
+	for file, dependencies := range dependencyGraph {
+		fmt.Printf("%s depends on:\n", strings.TrimPrefix(file, directory))
```

Comment:
nit: we need a better representation of dependencies. Please mark this as a TODO